Response model

acreilly opened the pull request:
To be merged after acr/pay_with_amazon branch
- Add SpreeAmazon::Response to help clean up the amazon gateway file

acreilly committed:
Add response model, utilize PayWithAmazon::Response, fixed specs

acreilly committed:
Add all success states

acreilly committed:
Reason code only shows on non success states

acreilly committed:
Specs for response model

mtomov commented on app/models/spree_amazon/response.rb:
```diff
@@ -0,0 +1,87 @@
+
+  def parse
+    Hash.from_xml(body)
+  end
```
This class really cleans up. The `Hash.from_xml` was very repetitive in the other PR. I'd even suggest to get rid of the `#parse` altogether, and always parse the `#body`, as why would you need just the body string?

acreilly replied:
For the other methods to work, it needs to be XML, since PayWithAmazon::Response has the method get_element to retrieve the response

mtomov replied:
Yeah, but you have wrapped that in a `#fetch`, which is internal to the `Response` class, so it has access to the raw XML:

```
  def fetch(path, element)
    response.get_element(path, element)
  end
```

acreilly replied:
If I do SpreeAmazon::Response::Authorization.new(mws_res) it returns the SpreeAmazon::Response::Authorization object, not the response.

mtomov commented on lib/amazon_mws.rb:
```diff
@@ -12,7 +12,7 @@
 class AmazonMwsOrderResponse
   def initialize(response)
-    @response = response.fetch("GetOrderReferenceDetailsResponse", {})
+    @response = Hash.from_xml(response.body).fetch("GetOrderReferenceDetailsResponse", {})
```
Shouldn't this be wrapped in the above class? Then you can use directly the `#body`, which will be parsed

acreilly replied:
How do you mean?

mtomov replied:
I mean `Response.new(response)` .. no?

acreilly replied:
GetOrderReferenceDetails model was never created though, I just didn't think it was worth it since there would have to be a lot of customization

acreilly replied:
https://github.com/boomerdigital/solidus_amazon_payments/blob/94059bad6a8dec11d7f5563576a2ea9c9046e11f/app/models/spree_amazon/response.rb#L79-L86

mtomov replied:
Maybe I'm just confused on this one. Only thing I was suggesting was to wrap the response in the newly created `Response` class, but maybe they are just different _responses_

acreilly replied:
I would have to create a class that inherits SpreeAmazon::Response called GetOrderReferenceDetails and do some customization since the GetOrderReferenceDetails response structure is different than other api calls

mtomov reviewed (commented):
Just glanced over the Response class, but once the other PR is merged, I'll look at this one again.

jordan-brough committed:
Update CodeClimate integration

After running a bundle update I was getting this message, which
aborted the specs:

> This usage of the Code Climate Test Reporter is now deprecated.
> Since version 1.0, we now require you to run `SimpleCov` in your
> test/spec helper, and then run the provided
> `codeclimate-test-reporter` binary separately to report your results
> to Code Climate.
>
> More information here:
> https://github.com/codeclimate/ruby-test-reporter/blob/master/README.md

jordan-brough committed:
empty - trigger specs

acreilly commented:
@mtomov @dhonig Probably merge this into acr/pay_with_amazon before merging that branch into master

acreilly committed:
Merge pull request #47 from jordan-brough/code-climate-update

Update CodeClimate integration

acreilly committed:
Merge branch 'master' into response-model

victorahm commented:
All good
